### DISCUSSION TITLE
Incomplete implementations?

### DISCUSSION THREAD

**Discussion by jtappin** [gnudatalanguage/gdl#861] [General]
A question for the developers.

Is there any point in contributing/adding library routines that don't include **all** of the features in the version in the IDL routine? 
Specifically I have an implementation of `r_correlate` that does not include the optional outputs `PROBD` and `ZD` as (a) I didn't need them  and (b) I'm not sure how to implement them. 

It needs a bit of work to integrate the Kendall coefficient which is currently a completely separate routine. It is actually much faster than IDL does it, and also gives correct answers for larger datasets.

**Comment by alaingdl** [comment]
Any contribution is welcome as long as it can be made under the GNU GPL licence. Some tests cases are also very welcome.

I would say incomplete code is not a problem, as long as the missing part is clearly visible
(clear warning/error messages). In the past I used to have interns to complete some code and missing functionalities ..
